test/test_torch.py

```diff
@@ -1763,6 +1763,33 @@ def test_randint_inference(self):
             self.assertIs(torch.int64, torch.randint(*args, size=size, out=out).dtype)
             self.assertIs(torch.int64, torch.randint(*args, size=size, out=out, dtype=torch.int64).dtype)
 
+    def test_full_inference(self):
+        size = (1, 2)
+
+        # Tests integer fill_values deprecated without specific dtype set
+        with self.maybeWarnsRegex(UserWarning, 'Deprecation warning: .+'):
+            torch.full(size, 1)
+        torch.full(size, 1, dtype=torch.int)
+
+        # Tests integer fill_values deprecated with named tensors, too
+        with self.maybeWarnsRegex(UserWarning, 'Deprecation warning: .+|Named tensors .+'):
+            torch.full(size, 1, names=('a', 'b'))
+        with self.maybeWarnsRegex(UserWarning, 'Named tensors .+'):
+            torch.full(size, 1, names=('a', 'b'), dtype=torch.float)
+
+        # Tests complex inference
+        self.assertTrue(torch.full(size, (1 + 1j)).dtype == torch.complex64)
```

Comment:
if you want to do this test now, you should probably test the difference default dtypes.

Reply:
oh, you do that below?  But why not with the annotation?

Reply:
There's not a great existing annotation for this since I'm setting torch.half as the default scalar type and the early part of the test is dtype agnostic. I can split the test and use the dtype annotation, however, which will make it less repetitive.